core/src/main/resources/lib/layout/icon.jelly

```diff
@@ -54,16 +54,23 @@ THE SOFTWARE.
     </j:when>
     <j:otherwise>
       <!-- Attempt to work back from the image url (backward compatibility)... -->
-      <j:set var="iconMetadata" value="${icons.getIconByUrl(imgSrc)}"/>
+      <j:set var="iconMetadata" value="${icons.getIconByUrl(iconSrc)}"/>
     </j:otherwise>
   </j:choose>
 
   <j:if test="${iconMetadata != null}">
-    <j:set var="imgSrc" value="${iconMetadata.getQualifiedUrl(context)}"/>
+    <j:set var="iconSrc" value="${iconMetadata.getQualifiedUrl(context)}"/>
     <j:set var="imgStyle" value="${iconMetadata.style} ${attrs.style}"/>
   </j:if>
 
-  <img class="${attrs.class}" src="${imgSrc}" style="${imgStyle}" title="${attrs.title}" height="${attrs.height}"
-       alt="${attrs.alt}" width="${attrs.width}" onclick="${attrs.onclick}" tooltip="${attrs.tooltip }"/>
-
-</j:jelly>
\ No newline at end of file
+  <j:choose>
+    <j:when test="${iconMetadata.svg}">
+      <l:svgIcon class="${attrs.class}" href="${iconSrc}"/>
```

Comment:
could probably support inline path as well, not sure if we need to though

Reply:
I added support for the format in the enum I think but not sure if we need it atm